### PR TITLE
Some bugs fixes for LR local predictions

### PR DESCRIPTION
**Pull request by felmarlop**
This PR fixes some bugs that I found while I implemented the LR local predictions. @joseribes helped me too.

Assigned to @mmerce

### CODE REVIEW

File: lib/LocalLogisticRegression.js [hunk 1]
```diff
@@ -450,7 +450,11 @@ LocalLogisticRegression.prototype.categoryProbability = function (
           index = this.items[fieldId].indexOf(term);
           occurrences = 1;
         }
-        probability += coefficients[shift + index] * occurrences;
+        var lookForIndex = shift + index;
```
Comment (mmerce) [inline lib/LocalLogisticRegression.js]: I don't think I totally understand this change. I guess you had problems when you try to predict with an input data which has an item that has never appeared in the training data. Then the 

``` javascript
index = this.items[fieldId].indexOf(term);
```

gives a negative index. Then I assume we should use the missing coefficient, but that's not the last one in the `coefficients` array... so are you really sure about this change? I mean, it looks like there's a bug in the old code, but I'm not sure it's really solved in your version... I might be mistaken though

File: lib/LocalLogisticRegression.js [hunk 2]
```diff
@@ -524,7 +528,7 @@ LocalLogisticRegression.prototype.getUniqueTerms = function (inputData) {
         inputData.hasOwnProperty(fieldId)) {
       regexp = utils.separatorRegexp(this.itemAnalysis[fieldId]);
       inputDataField = "";
-      inputDataField = inputData[fieldId];
+      inputDataField = inputData[fieldId].toString();
```
Comment (mmerce) [inline lib/LocalLogisticRegression.js]: does line 511 in your new code need to be changed too?

File: lib/LocalLogisticRegression.js (hunk 1 above)
Comment (felmarlop) [inline lib/LocalLogisticRegression.js]: Im going to check again and I tell you

File: lib/LocalLogisticRegression.js (hunk 2 above)
Comment (felmarlop) [inline lib/LocalLogisticRegression.js]: That is the case where I found no-text item and split function didn't work.

**Comment by mmerce** [comment]
Changes included in commit 36f19f5a7c93fe0dd2db2c5d12642ed52747694a. Thanks to @felmarlop and @joseribes for improvements.

**Pull request closed without merging**